File: BehavioralDesignPattern/MementoPattern/python/Driver.py
from TextEditor import TextEditor
from CareTaker import CareTaker

if __name__ == "__main__":
    TextEditor = TextEditor()
    CareTaker = CareTaker()

    TextEditor.setContent("Hello from 1st content")
    CareTaker.save(TextEditor)
    print("Content after first save: {}".format(TextEditor.getContent()))

    TextEditor.setContent("Hello from 2nd content")
    print("Content after second text change: {}".format(TextEditor.getContent()))

    CareTaker.undo(TextEditor)
    print("Content after first undo: {}".format(TextEditor.getContent()))

    CareTaker.undo(TextEditor)
    print("Content after second undo: {}".format(TextEditor.getContent()))

File: BehavioralDesignPattern/MementoPattern/java/Driver.java
package BehavioralDesignPattern.MementoPattern.java;

public class Driver {
    public static void main(String[] args){
        TextEditor textEditor = new TextEditor();
        CareTaker careTaker = new CareTaker();
        textEditor.setContent("Hello 1 friend");
        careTaker.save(textEditor);
        System.out.println("Content after first save: "+textEditor.getContent());

        textEditor.setContent("Hello  2 friends");
        System.out.println("Content after second change: "+textEditor.getContent());

        careTaker.undo(textEditor);
        System.out.println("Content after first undo: "+textEditor.getContent());

        careTaker.undo(textEditor);
        System.out.println("Content after second undo: "+textEditor.getContent());
    }
}
